File: Backend/Models/Paiement.cs
namespace ProjetV1.Models
{
    public class Paiement
    {
        public int Id { get; set; }
        public int TicketId { get; set; }
        public Ticket Ticket { get; set; }
        public int UtilisateurId { get; set; }
        public Utilisateur Utilisateur { get; set; }
        public string ModePaiment { get; set; }
        public string Statut { get; set; } // valide ou refuse
    }
}

File: backend/app/Models/Paiement.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Paiement extends Model
{
    use HasFactory;

    protected $fillable = [
        'ticket_id',
        'user_id',
        'mode_paiement',
        'statut',
        'verification_code',
    ];

    // Relation to Ticket
    public function ticket()
    {
        return $this->belongsTo(Ticket::class, 'ticket_id');
    }


    public function user()
    {
        return $this->belongsTo(User::class, 'user_id');
    }
}
